object-filter/src/main/java/org/infinispan/objectfilter/impl/aggregation/SumAccumulator.java

```diff
@@ -0,0 +1,25 @@
+package org.infinispan.objectfilter.impl.aggregation;
+
+/**
+ * @author anistor@redhat.com
+ * @since 8.0
+ */
+public final class SumAccumulator extends FieldAccumulator {
+
+   public SumAccumulator(int pos) {
+      super(pos);
+   }
+
+   @Override
+   public void update(Object[] srcRow, Object[] destRow) {
+      Number value = (Number) srcRow[pos];
+      if (value != null) {
+         Number sum = (Number) destRow[pos];
+         if (sum == null) {
+            destRow[pos] = value;
+         } else {
+            destRow[pos] = sum.intValue() + value.intValue(); //todo [anistor] this handles integers only!
```

Comment:
I've updated the accumulator to correctly compute sums based on the type of the property. Byte and short are promoted to int as it normally happens in java, the others remain unchanged.